app/test/unit/git/cherry-pick-test.ts

```diff
@@ -0,0 +1,250 @@
+    // cherry pick and will check out the target branch.
+    repository = await createRepository(featureBranchName, targetBranchName)
+
+    // branch with tip as commit to cherry pick
+    featureBranch = await getBranchOrError(repository, featureBranchName)
+
+    // branch with to cherry pick to
+    targetBranch = await getBranchOrError(repository, targetBranchName)
+
+    // set result to null for each test
+    result = null
+  })
+
+  it('successfully cherry pick one commit without conflicts', async () => {
+    result = await cherryPick(repository, featureBranch.tip.sha)
+    const cherryPickedCommit = await getCommit(
+      repository,
+      featureBranch.tip.sha
+    )
+
+    const commits = await getCommits(repository, targetBranch.ref, 3)
+    expect(commits.length).toBe(2)
+    expect(commits[0].summary).toBe(cherryPickedCommit!.summary)
+    expect(result).toBe(CherryPickResult.CompletedWithoutError)
+  })
+
+  it('successfully cherry pick a commit with empty message', async () => {
+    // add a commit with no message
+    await switchTo(repository, featureBranchName)
+    const filePath = Path.join(repository.path, 'EMPTY_MESSAGE.md')
+    await FSE.writeFile(filePath, '# HELLO WORLD! \nTHINGS GO HERE\n')
+    await GitProcess.exec(['add', filePath], repository.path)
+    await GitProcess.exec(
+      ['commit', '--allow-empty-message', '-m', ''],
+      repository.path
+    )
+
+    featureBranch = await getBranchOrError(repository, featureBranchName)
+    await switchTo(repository, targetBranchName)
+
+    result = await cherryPick(repository, featureBranch.tip.sha)
+    const commits = await getCommits(repository, targetBranch.ref, 5)
+    expect(commits.length).toBe(2)
+    expect(commits[0]!.summary).toBe('')
+    expect(result).toBe(CherryPickResult.CompletedWithoutError)
+  })
+
+  it('successfully cherry pick multiple commits without conflicts', async () => {
+    // keep reference to the first commit in cherry pick range
+    const firstCommitSha = featureBranch.tip.sha
+
+    // add two more commits to cherry pick
+    await switchTo(repository, featureBranchName)
+
+    const featureBranchCommitTwo = {
+      commitMessage: 'Cherry Picked Feature! Number Two',
+      entries: [
+        {
+          path: 'THING_TWO.md',
+          contents: '# HELLO WORLD! \nTHINGS GO HERE\n',
+        },
+      ],
+    }
+    await makeCommit(repository, featureBranchCommitTwo)
+
+    const featureBranchCommitThree = {
+      commitMessage: 'Cherry Picked Feature! Number Three',
+      entries: [
+        {
+          path: 'THING_THREE.md',
+          contents: '# HELLO WORLD! \nTHINGS GO HERE\n',
+        },
+      ],
+    }
+    await makeCommit(repository, featureBranchCommitThree)
+
+    featureBranch = await getBranchOrError(repository, featureBranchName)
+    await switchTo(repository, targetBranchName)
+
+    const commitRange = `${firstCommitSha}^..${featureBranch.tip.sha}`
```

Comment:
:D - I was going to look to see if we had something like this when I went to implement range cherry picking ui.  I will go ahead and add it since I will need for that later PR anyways.